### PR TITLE
🐛 [PRTL-1847] ignored id with invalid characters when validating

### PR DESCRIPTION
**Pull request by aishmitk**

### CODE REVIEW

File: src/packages/@ncigdc/components/Modals/UploadSet/Base.js [hunk 1]
```diff
@@ -24,15 +25,10 @@ const enhance = compose(
...
-          ...(/^[a-zA-Z0-9\->:]*$/.test(id)
-            ? { noSpecialCharHits: [...acc.noSpecialCharHits, id] }
-            : { specialCharHits: [...acc.specialCharHits, id] }),
-        }),
-        { noSpecialCharHits: [], specialCharHits: [] },
-      );
+      const {
+        true: noSpecialCharHits = [],
+        false: specialCharHits = [],
+      } = groupBy(hits, id => /^[a-zA-Z0-9\->:]*$/.test(id));
```
Comment (cy) [inline src/packages/@ncigdc/components/Modals/UploadSet/Base.js]: could use a ternary to assign the keys
```Javascript
        const { noSpecialCharHits = [], specialCharHits = [] } = groupBy(
          hits,
          id =>
            /^[a-zA-Z0-9\->:]*$/.test(id)
              ? 'noSpecialCharHits'
              : 'specialCharHits',
        );
```

File: src/packages/@ncigdc/utils/validateIds.js [hunk 2]
```diff
@@ -75,24 +75,26 @@ export const validateCases = validate(
 function validate(idFields, map, field, endpoint, extraFields = []) {
   return async (ids = [], onValidatingStateChange) => {
     const notValidatedIds = ids.filter(g => typeof map[g] === 'undefined');
-    if (notValidatedIds.length <= 0) return;
+    const filteredUnvalidatedIds = notValidatedIds.filter(x =>
```
Comment (cy) [inline src/packages/@ncigdc/utils/validateIds.js]: this file can be reset

**Comment by aishmitk** [comment]
that is one rad emoji

**Comment by alex-wilmer** [comment]
> reverted to old code

wrong branch?

**Comment by aishmitk** [comment]
yep

**Comment by alex-wilmer** [comment]
wait closed? wasn't this still valid before

**Comment by aishmitk** [comment]
mean to click comment and click 'close and comment' 😪

**Comment by alex-wilmer** [comment]
haha okay

**Comment by cy** [comment]
let's squash & merge this thing :shipit:

**Pull request merged**